File: creational/java/builder/src/CarBuilder.java
public abstract class CarBuilder {
    protected Car car;

    public void createDefaultCar(){ this.car = new Car(); }
    public Car getCar(){ return this.car; }
    public abstract void buildEngine();
    public abstract void buildHeadLight();
    public abstract void buildInterior();
    public abstract void buildSunroof();
}

File: creational/python/builder/CarBuilder.py
from abc import ABC, abstractmethod
from Car import Car

class CarBuilder(ABC):
    _car: Car = None

    def createDefaultCar(self)->None:
        self._car = Car()

    def getCar(self)->Car:
        return self._car

    @abstractmethod
    def buildEngine(self)->None:
        pass

    @abstractmethod
    def buildHeadLight(self)->None:
        pass

    @abstractmethod
    def buildInterior(self)->None:
        pass

    @abstractmethod
    def buildSunroof(self) -> None:
        pass